### DISCUSSION TITLE
How categorical feature gets handle by FLAML

### DISCUSSION THREAD

**Discussion by anandece4u** [microsoft/FLAML#124] [Q&A]
Hi ,
I have a dataset with customer_id as a feature and in my current catboost classification model I am using that feature as a categorical feature by passing that column into "cat_features" parameter.
My question is how can I use FLAML for above dataset where I want my customer_id as categorical feature. Will FLAML handle categorical feature automatically or there is some work around for that?

**Comment by sonichi** [comment]
Hi @anandece4u 
If you use pandas dataframe and the type for the customer_id column is 'object' or 'category', then it will be treated as a categorical feature automatically. Would that work for you?

**Reply by anandece4u** [reply]
Yes absolutely works for me. Thank you so much for quick response. 
But I am just curious to know how exactly it handles object type like one hot encoding or label encoding etc. Because you know Catboost handles categorical feature differently, so Catboost used inside FLAML uses the same encoding like original Catboost or its different in FLAML?

**Reply by sonichi** [reply]
Different estimators use different encodings. Catboost used inside FLAML uses the same encoding like original Catboost. That can be found in the `_preprocess` function of each estimator in `model.py`.

* LGBM and Catboost's preprocess doesn't do anything to categorical features, effectively leaving them "raw"
* KNN's preprocess drops all categorical features
* Other estimators use a Ordinal-type encoding via `X[cat_columns] = X[cat_columns].apply(lambda x: x.cat.codes)`

Related: https://github.com/microsoft/FLAML/issues/117

**Reply by anandece4u** [reply]
Thanks a lot for detailed answer. 
It's really helpful and at the same time clear all my doubts also. thanks
